applications/OptimizationApplication/python_scripts/algorithms/algorithm_relaxed_gradient_projection.py

```diff
@@ -0,0 +1,317 @@
+            return w_r, w_c
+
+    @time_decorator()
+    def ComputeControlUpdate(self, alpha) -> KratosOA.CollectiveExpression:
+        search_direction = self.algorithm_data.GetBufferedData()["search_direction"]
+        update = KratosOA.ExpressionUtils.Scale(search_direction, alpha)
+        self.algorithm_data.GetBufferedData().SetValue("control_field_update", update.Clone(), overwrite=True)
+        return update
+
+    @time_decorator()
+    def UpdateControl(self) -> KratosOA.CollectiveExpression:
+        update = self.algorithm_data.GetBufferedData()["control_field_update"]
+        self.__control_field += update
+
+    @time_decorator()
+    def GetCurrentObjValue(self) -> float:
+        return self.__obj_val
+
+    @time_decorator()
+    def GetCurrentControlField(self):
+        return self.__control_field
+
+    @time_decorator()
+    def Output(self) -> KratosOA.CollectiveExpression:
+        self.algorithm_data.GetBufferedData()["control_field"] = self.__control_field.Clone()
+        for process in self._optimization_problem.GetListOfProcesses("output_processes"):
+            if process.IsOutputStep():
+                process.PrintOutput()
+
+    def CheckLinearizedConstraints(self, active_constr_grad: 'list[KratosOA.CollectiveExpression]', update: KratosOA.CollectiveExpression, w_r, w_c) -> bool:
+        all_feasible = True
+        active_constraints_list = [self.__constraints_list[i] for i in range(len(self.__constraints_list)) if self.__constraints_list[i].IsActiveConstrant()]
+        for i, constraint in enumerate(active_constraints_list):
+            predicted_value = constraint.GetStandardizedValue() + KratosOA.ExpressionUtils.InnerProduct(active_constr_grad[i], update)
+            print(f"RGP Constaint {constraint.GetResponseName()}:: predicted g_i {predicted_value}")
+            if not constraint.IsSatisfied(predicted_value):
+                all_feasible = False
+                w = constraint.ComputeW()
+                w += self.buffer_coeff_update_factor
+                w_r[i] = constraint.Compute_W_relax(w)
+                w_c[i] = constraint.Compute_W_correction(w)
+                print(f"RGP Constaint {constraint.GetResponseName()}:: Corrected w_r = {w_r[i]}, w_c = {w_c[i]}")
+            else:
+                print(f"RGP Constaint {constraint.GetResponseName()}:: no correction for w_r, w_c ")
+        return all_feasible
+
+
+    @time_decorator()
+    def Solve(self):
+        while not self.converged:
+            with OptimizationAlgorithmTimeLogger("Gradient Projection",self._optimization_problem.GetStep()):
+                self._InitializeIteration()
+
+                self.__obj_val = self.__objective.CalculateStandardizedValue(self.__control_field)
+                obj_info = self.__objective.GetInfo()
+                self.algorithm_data.GetBufferedData()["std_obj_value"] = obj_info["value"]
```

Comment:
As far as I see, the `obj_info["value"]` only returns the objective value, not the standardized objective value. Could you check it?

Reply:
you are right, fixed.